packages/components/src/progressive-disclosure-panel/index.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import noop from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ProgressiveDisclosurePanelTitle from './title';
+
+const ProgressiveDisclosurePanel = ( props ) => {
+	const { children, className, label: menuLabel, resetAll, title } = props;
+	const [ menuItems, setMenuItems ] = useState( {} );
+	const [ defaultChildren, setDefaultChildren ] = useState( {} );
```

Comment:
I'm not sure that the proposed simplification would work with the current requirements. Its quite possible I'm misunderstanding the issue though.

The current `menuItems` state can represent everything that is needed to generate the menu, including whether an item is checked or not. I think the proposed simplification would move some of the complexity then to menu generation as well as potentially making that state harder to reason about at a glance ( at least it does for me 🙂 ).

Your point about simplifying the state though is well received. To that end and following on from your earlier suggestions and comments around the management of the default controls state, we could possibly remove that. It could be moved into a hook such that it is more robust in the face of any use case where the ProgressiveDisclosurePanel totally changed the panel's children after initial render.

I'll clean this up, then it can be revisited when the picture is a little bit clearer.